numpy/core/src/umath/_umath_tests.c.src

```diff
@@ -402,12 +422,15 @@ init_umath_tests(void)
     Py_DECREF(version);
 
     /* Load the ufunc operators into the module's namespace */
-    addUfuncs(d);
-
-    if (PyErr_Occurred()) {
+    if (addUfuncs(d) < 0) {
+        Py_DECREF(m);
+        if (PyErr_Occurred()) {
```

Comment:
Either:

* This test is pointless
* There is a bug in `PyUFunc_FromFuncAndDataAndSignature` which fails to set an exception correctly

I'm assuming the former, meaning you can just call `PyErr_Print()` unconditionally

Reply:
I suppose my point is that `addUfuncs() < 0` is already our test to be sure that an error is set. If we returned -1 but forgot to set an error, that's a bug.

Also it seems that [`PyErr_Print` is perfectly ok with there being no error anyway](https://github.com/python/cpython/blob/42aa93b8ff2f7879282b06efc73a31ec7785e602/Python/pythonrun.c#L640-L650). So I'd say keep the PyErr_Print(), but drop the condition guarding it.